lib/xfrm/sa.c

```diff
@@ -191,7 +191,7 @@ static int xfrm_sa_clone(struct nl_object *_dst, struct nl_object *_src)
 
 	if (src->sec_ctx)
 	{
-		len = sizeof (struct xfrmnl_sec_ctx) + src->sec_ctx->ctx_len;
+		len = sizeof (src->sec_ctx) + src->sec_ctx->ctx_len;
```

Comment:
You're right, good catch. This one slipped through my test program on a 64 bit machine, since sizeof(uintptr_t) is equal to sizeof(struct xfrmnl_user_sec_ctx).